lib/chitter.rb

```diff
@@ -0,0 +1,35 @@
+require 'pg'
+require 'uri'
+require_relative 'user'
+require_relative 'database_connection'
+
+class Chitter
+  attr_reader :id, :message, :nickname, :post_time
+
+  def initialize(id:, message:, nickname:, post_time:)
+    @id = id
+    @message = message
+    @nickname = nickname
+    @post_time = post_time
+  end
+
+  def self.open
+    DatabaseConnection.query("SELECT * FROM messages ORDER BY post_time DESC;")
+  end
+
+  def self.create(nickname, message, post_time)
+    result = DatabaseConnection.query(
+      "INSERT INTO messages
+      (nickname, message, post_time)
+      VALUES('#{nickname}','#{message}','#{post_time}')
+      RETURNING id, nickname, message, post_time;"
+    )
+    Chitter.new(
+      id: result[0]['id'],
```

Comment:
maybe it would make your code DRYer if you used .first on result above, rather than having to use [0] here?